src/elements/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+}
+
+function flattenNodes(node: TreeElementNode, collection: TreeElementNode[]) {
+    if (node.checked) {
+        collection.push(node);
+    }
+    if (node.children && node.children.length > 0) {
+        for (const child of node.children) {
+            flattenNodes(child, collection);
+        }
+    } else {
+        return;
+    }
+}
+
+function filterTree(nodes: TreeElementNode[]) {
+    const children: TreeElementNode[] = [];
+    for (const node of nodes) {
+        flattenNodes(node, children);
+    }
+    return children;
+}
+
+function renderChild(
+    node: TreeElementNode,
+    index: number,
+    value: TreeElementNode[] | undefined,
+    showCheckBoxes: boolean | undefined,
+    level: number,
+    brandingService: BrandingService,
+    handleMouseEnter: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleMouseLeave: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleItemClick: (node: any) => void,
+    handleFolderClick: (node: any) => void,
+    handleSecondaryActionClick: (event: React.MouseEvent, node: TreeElementNode) => void
+) {
+    const isDense = brandingService.density === "compact";
+    const isFolder = Array.isArray(node.children);
+    const isNode = !node.children;
+    const m = 1.5 * level;
+    const icon = (
+        <ChevronDownIcon
+            fontSize="small"
+            className="gcx-layer-tree-node-expand-button"
+            sx={{
+                transform: `rotate(${node.open ? 0 : -90}deg)`,
+                transition: "transform 0.2s ease-in-out",
+            }}
+        />
+    );
+    return (
+        <React.Fragment>
+            <ListItemButton
+                key={node.key}
+                className={clsx(FOCUSABLE_LIST_ITEM_CLASS, {
+                    "gcx-layer-tree-node-compact": isDense,
+                })}
+                sx={{ paddingLeft: (theme) => theme.spacing(m) }}
+                onMouseEnter={(event) => handleMouseEnter(event, node)}
+                onMouseLeave={(event) => handleMouseLeave(event, node)}
+            >
+                <ListItemIcon
+                    className="layer-tree-item-icon-container"
+                    sx={{ alignItems: "center", minWidth: 70 }}
+                    data-testid="layer-tree-item-icon-container"
+                >
+                    {isNode && showCheckBoxes && (
+                        <Checkbox
+                            className="gcx-layer-tree-node-checked-toggle"
+                            checked={node.checked}
+                            tabIndex={-1}
+                            disableRipple
+                            onClick={() => handleItemClick(node)}
+                        />
+                    )}
+                    {node.icon && <DynamicIcon src={node.icon} fontSize="small" sx={{ ml: 1 }} />}
+                </ListItemIcon>
+
+                {node.link ? (
+                    <Link rel="noreferrer" target="_blank" href={node.link}>
```

Comment:
I'll be honest... I hate the `link`. We are trying to do too much. If the click behavior should open a link we should do that via the `onClick` handler and not bake it into the component.

Reply:
Everything now is a button and it's up to the implementation to decide what happens on the secondary click event.